### PR TITLE
[Internal] Hedging: Refactors Tests 

### PR DESCRIPTION
**Pull request by NaluTripician**
# Pull Request Template

## Description

Test improvements to decrease test time for hedging. Includes having a set database that is to be reused test to test, and disposing clients properly + stopping the change feed. 

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosAvailabilityStrategyTests.cs [hunk 1]
```diff
@@ -152,20 +154,35 @@ public async Task TestInitAsync()
...
+            {
+                //Test to see if the container exists
+                //will return a 404 1003 if it does not which inidcates we need to create test resources
+                await this.container.ReadThroughputAsync();
+            }
+            catch (CosmosException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
+            {
+                //if test db or container does not exist create them
+                this.database = await this.client.CreateDatabaseIfNotExistsAsync(dbName);
+                this.container = await this.database.CreateContainerIfNotExistsAsync(containerName, "/pk");
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosAvailabilityStrategyTests.cs:171]: Just calling createIfNotExists might suffice assuming the status codes can be used to distinguish between created vs read in response, thoughts?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosAvailabilityStrategyTests.cs [hunk 2]
```diff
@@ -152,20 +154,35 @@ public async Task TestInitAsync()
...
-            await Task.Delay(60000);
+            try
+            {
+                //Test to see if the container exists
+                //will return a 404 1003 if it does not which inidcates we need to create test resources
+                await this.container.ReadThroughputAsync();
+            }
+            catch (CosmosException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
+            {
+                //if test db or container does not exist create them
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosAvailabilityStrategyTests.cs:169]: Const or static convention is to use with the className.property

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosAvailabilityStrategyTests.cs (hunk 1 above)
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosAvailabilityStrategyTests.cs:171]: Default RUs: Lets define it to for COGS (defaults are expensive)

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosAvailabilityStrategyTests.cs [hunk 3]
```diff
@@ -152,20 +154,34 @@ public async Task TestInitAsync()
...
+                    new AvailabilityStrategyTestObject { Id = "testId", Pk = "pk" });
+                await this.container.CreateItemAsync<AvailabilityStrategyTestObject>(
+                    new AvailabilityStrategyTestObject { Id = "testId2", Pk = "pk2" });
+                await this.container.CreateItemAsync<AvailabilityStrategyTestObject>(
+                    new AvailabilityStrategyTestObject { Id = "testId3", Pk = "pk3" });
+                await this.container.CreateItemAsync<AvailabilityStrategyTestObject>(
+                    new AvailabilityStrategyTestObject { Id = "testId4", Pk = "pk4" });
+
+                //Must Ensure the data is replicated to all regions
+                await Task.Delay(60000);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosAvailabilityStrategyTests.cs:183]: Is Delay needs to be conditional only for container create? (database might exisit but not container)

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosMultiRegionDiagnosticsTests.cs [hunk 4]
```diff
@@ -2,47 +2,67 @@
...
         {
             this.connectionString = ConfigurationManager.GetEnvironmentVariable<string>("COSMOSDB_MULTI_REGION", null);
             this.client = new CosmosClient(this.connectionString);
 
-            this.dbName = Guid.NewGuid().ToString();
-            this.database = await this.client.CreateDatabaseIfNotExistsAsync(this.dbName);
-
-            this.containerName = Guid.NewGuid().ToString();
-            this.container = await this.database.CreateContainerIfNotExistsAsync(this.containerName, "/pk");
+            DatabaseResponse db = await this.client.CreateDatabaseIfNotExistsAsync(
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosMultiRegionDiagnosticsTests.cs:32]: Thoughts on refracturing as a utility function?

**Review by kirankumarkolli — DISMISSED** [review]

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Utils/MultiRegionSetupHelpers.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                container = await database.CreateContainerIfNotExistsAsync(
+                    id: MultiRegionSetupHelpers.containerName,
+                    partitionKeyPath: "/pk",
+                    throughput: 400);
+                changeFeedContainer = await database.CreateContainerIfNotExistsAsync(
+                    id: MultiRegionSetupHelpers.changeFeedContainerName,
+                    partitionKeyPath: "/partitionKey",
+                    throughput: 400);
+
+                await container.CreateItemAsync<AvailabilityStrategyTestObject>(
```
Comment (kundadebdatta) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Utils/MultiRegionSetupHelpers.cs:39]: Thoughts on running the tasks in parallel ?

```
List<Task> tasks = new List<Task>
{
    container.CreateItemAsync<AvailabilityStrategyTestObject>(
            new AvailabilityStrategyTestObject { Id = "testId", Pk = "pk" }),
    container.CreateItemAsync<AvailabilityStrategyTestObject>(
        new AvailabilityStrategyTestObject { Id = "testId2", Pk = "pk2" }),
    container.CreateItemAsync<AvailabilityStrategyTestObject>(
        new AvailabilityStrategyTestObject { Id = "testId3", Pk = "pk3" }),
    container.CreateItemAsync<AvailabilityStrategyTestObject>(
        new AvailabilityStrategyTestObject { Id = "testId4", Pk = "pk4" })                                                             
};

await Task.WhenAll(tasks);
```

**Review by kundadebdatta — APPROVED** [review]
LGTM!

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**